crates/re_chunk/src/batcher.rs

```diff
@@ -0,0 +1,1588 @@
+    ///
+    /// See [`Self::apply_env`].
+    #[inline]
+    pub fn from_env() -> ChunkBatcherResult<Self> {
+        Self::default().apply_env()
+    }
+
+    /// Returns a copy of `self`, overriding existing fields with values from the environment if
+    /// they are present.
+    ///
+    /// See [`Self::ENV_FLUSH_TICK`], [`Self::ENV_FLUSH_NUM_BYTES`], [`Self::ENV_FLUSH_NUM_BYTES`].
+    pub fn apply_env(&self) -> ChunkBatcherResult<Self> {
+        let mut new = self.clone();
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_TICK) {
+            let flush_duration_secs: f64 =
+                s.parse().map_err(|err| ChunkBatcherError::ParseConfig {
+                    name: Self::ENV_FLUSH_TICK,
+                    value: s.clone(),
+                    err: Box::new(err),
+                })?;
+
+            new.flush_tick = Duration::from_secs_f64(flush_duration_secs);
+        }
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_NUM_BYTES) {
+            if let Some(num_bytes) = re_format::parse_bytes(&s) {
+                // e.g. "10MB"
+                new.flush_num_bytes = num_bytes.unsigned_abs();
+            } else {
+                // Assume it's just an integer
+                new.flush_num_bytes = s.parse().map_err(|err| ChunkBatcherError::ParseConfig {
+                    name: Self::ENV_FLUSH_NUM_BYTES,
+                    value: s.clone(),
+                    err: Box::new(err),
+                })?;
+            }
+        }
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_NUM_ROWS) {
+            new.flush_num_rows = s.parse().map_err(|err| ChunkBatcherError::ParseConfig {
+                name: Self::ENV_FLUSH_NUM_ROWS,
+                value: s.clone(),
+                err: Box::new(err),
+            })?;
+        }
+
+        if let Ok(s) = std::env::var(Self::ENV_MAX_CHUNK_ROWS_IF_UNSORTED) {
+            new.max_chunk_rows_if_unsorted =
+                s.parse().map_err(|err| ChunkBatcherError::ParseConfig {
+                    name: Self::ENV_MAX_CHUNK_ROWS_IF_UNSORTED,
+                    value: s.clone(),
+                    err: Box::new(err),
+                })?;
+        }
+
+        Ok(new)
+    }
+}
+
+#[test]
+fn chunk_batcher_config() {
+    // Detect breaking changes in our environment variables.
+    std::env::set_var("RERUN_FLUSH_TICK_SECS", "0.3");
+    std::env::set_var("RERUN_FLUSH_NUM_BYTES", "42");
+    std::env::set_var("RERUN_FLUSH_NUM_ROWS", "666");
+    std::env::set_var("RERUN_MAX_CHUNK_ROWS_IF_UNSORTED", "7777");
+
+    let config = ChunkBatcherConfig::from_env().unwrap();
+
+    let expected = ChunkBatcherConfig {
+        flush_tick: Duration::from_millis(300),
+        flush_num_bytes: 42,
+        flush_num_rows: 666,
+        max_chunk_rows_if_unsorted: 7777,
+        ..Default::default()
+    };
+
+    assert_eq!(expected, config);
+}
+
+// ---
+
+/// Implements an asynchronous batcher that coalesces [`PendingRow`]s into [`Chunk`]s based upon
```

Comment:
This docstring seems like it should contain the invariants from the PR description.